tests/db.rs

```diff
@@ -0,0 +1,184 @@
+use rand::rngs::StdRng;
+use rand::{Rng, SeedableRng};
+use slatedb::config::{CompactorOptions, DbOptions, PutOptions, WriteOptions};
+use slatedb::object_store::memory::InMemory;
+use slatedb::object_store::path::Path;
+use slatedb::object_store::ObjectStore;
+use slatedb::Db;
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+async fn test_concurrent_writers_and_readers() {
+    const NUM_WRITERS: usize = 10;
+    const NUM_READERS: usize = 2;
+    const WRITES_PER_TASK: usize = 100;
+    const KEY_LENGTH: usize = 256;
+
+    // Pad keys to allow us to control how many blocks we take up
+    // Since block size is not configurable
+    fn zero_pad_key(key: u64, length: usize) -> Vec<u8> {
+        let mut padded_key = vec![0; length];
+        // Set the first 8 bytes to the key
+        padded_key[0..8].copy_from_slice(&key.to_le_bytes());
+        padded_key
+    }
+
+    // Create an InMemory object store and DB
+    let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
+    let config = DbOptions {
+        flush_interval: Some(Duration::from_millis(100)),
+        manifest_poll_interval: Duration::from_millis(100),
+        compactor_options: Some(CompactorOptions {
+            poll_interval: Duration::from_millis(100),
+            ..Default::default()
+        }),
+        // Allow 16KB of unflushed data
+        max_unflushed_bytes: 16 * 1024,
```

Comment:
Should there be asserts on the kind of SST files produced (number of blocks, number of files etc)? 
As the implementation evolves, these things could change and the test may not be validating what it does today.

Reply:
My concern is that the test is not deterministic. The best I think I could do would be to verify that at least one L0 SST was created with multiple blocks. Even then, I can't easily guarantee that reads hit that SST.

As it is, the test will yield false negatives but not false positives. That is, it will only fail if something is actually wrong. If I add asserts, it could fail if nothing is wrong except we got unlucky with the non-determinism.